specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json

```diff
@@ -14719,6 +14784,39 @@
         }
       }
     },
+    "RecoveryPlanProviderSpecificDetails": {
+      "description": "Recovery plan provider specific details.",
+      "type": "object",
+      "properties": {
+        "instanceType": {
+          "description": "Gets the Instance type.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "discriminator": "instanceType"
+    },
+    "RecoveryPlanProviderSpecificInput": {
+      "description": "Recovery plan provider specific input base class.",
+      "type": "object",
+      "properties": {
+        "instanceType": {
+          "description": "Gets the Instance type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "canWrite": {
```

Comment:
canRead\Write are set to readonly but only used as inputs to an API, how did you intend to use them? As they are currently modeled, you should remove them.